doc/en/deprecations.rst

```diff
@@ -39,9 +39,28 @@ Node Construction changed to ``Node.from_parent``
 
 .. deprecated:: 5.4
 
-The construction of nodes new should use the named constructor ``from_parent``.
+The construction of nodes now should use the named constructor ``from_parent``.
 This limitation in api surface intends to enable better/simpler refactoring of the collection tree.
 
+This means that instead of :code:`MyItem(name="foo", parent=collector, obj=42)`
+one now has to invoke :code:`MyItem.from_parent(collector, name="foo")`.
+
+Plugins that wish to support older versions of pytest and suppress the warning can use
+`hasattr` to check if `from_parent` exists in that version:
+
+.. code-block:: python
+
+    def pytest_pycollect_makeitem(collector, name, obj):
+        if hasattr(MyItem, "from_parent"):
+            item = MyItem.from_parent(collector, name="foo")
+            item.obj = 42
```

Comment:
This seems a bit strange to me... how are users supposed to write their constructors?

```python
class MyItem(Node):

    def __init__(self, parent, name):
        super().__init__(parent=parent, name=name)
        self.obj = None


# constructed like this:
item = MyItem.from_parent(collector, name="foo")
item.obj = 42
```

Or are they supposed to override `from_parent`?

```python
class MyItem(Node):

    def __init__(self, parent, name, obj):
        super().__init__(parent=parent, name=name)
        self.obj = obj

    @classmethod
    def from_parent(cls, parent, name, obj):
        return super().from_parent(parent=parent, name=name, obj=obj) 

# constructed like this:
item = MyItem.from_parent(collector, name="foo", obj=42)
```

?

Reply:
for initially users will have to override both, we will eventually migrate to a model where they only override from_parent

Reply:
> for initially users will have to override both

So I assume you mean the latter sample, right?

> we will eventually migrate to a model where they only override from_parent

But how is that possible (to only override `from_parent`)? Users still need to define their attributes somewhere. 

How to you see this playing out in the future? 

Also why users even need to implement `from_parent` at all, given that the framework doesn't call that function? Seems weird to have as prerequisite to override `__init__` (which is usually OK) and then `from_parent`, which pytest won't call at all: the user is responsible for creating nodes on the impl of the appropriate hooks (`pytest_pycollect_makeitem`).

Reply:
Should we update the suggestion to go for `attr.s` directly, and implementing their own `from_parent` class?

Reply:
Nevermind, I realize this won't work with `attr.s` because of the superclass relationship.